tests/utils.py

```diff
@@ -13,21 +13,20 @@
 import unittest
 from unittest import mock
 
-import freezegun
 import httpx
 import respx
+import time_machine
 
 from cumulus_etl.formats.deltalake import DeltaLakeFormat
 
-# Pass a non-UTC time to freezegun to help notice any bad timezone handling.
+# Pass a non-UTC time to time-machine to help notice any bad timezone handling.
 # But only bother exposing the UTC version to other test code, since that's what will be most useful/common.
 _FROZEN_TIME = datetime.datetime(2021, 9, 15, 1, 23, 45, tzinfo=datetime.timezone(datetime.timedelta(hours=4)))
 FROZEN_TIME_UTC = _FROZEN_TIME.astimezone(datetime.timezone.utc)
 
 
 # Several tests involve timestamps in some form, so just pick a standard time for all tests.
-# We ignore socketserver because it checks the result of time() when evaluating timeouts.
-@freezegun.freeze_time(_FROZEN_TIME, ignore=["socketserver"])
+@time_machine.travel(_FROZEN_TIME, tick=False)
```

Comment:
This switch fixes a crash I was seeing between pandas and freezegun (pandas only likes to use real DateTime objects, not fake ones and freezegun was generating mocked objects). Time Machine instead changes things at a lower level, and pandas seems happy.

It also fixes the issue with `socketserver` magically (despite time not ticking forward.... :shrug:)

Reply:
oh that's interesting - i've had to jump in hoops around that, i'll look into it.